src/match/macro.jl

```diff
@@ -69,11 +69,11 @@ macro capture(ex, pat)
   pat = subtb(subor(pat))
   quote
     $([:($(esc(b)) = nothing) for b in bs]...)
-    env = trymatch($(Expr(:quote, pat)), $(esc(ex)))
+    env = trymatch($(esc(Expr(:quote, pat))), $(esc(ex)))
```

Comment:
I don't understand how come it worked before. Was `trymatch` written loosely to account for the possibility of hygiene mangling the pattern? I don't use hygiene so I'm unfamiliar with its details.

Reply:
Or wait - does hygiene even process quoted expressions? If so, why is the `esc` necessary?

Reply:
`@capture` is usually used in a following way:
```julia
@capture(:(sin(1)), f_(a_))
```
In this example, `f_(a_)` is passed as a `Symbol` literal and so we don't need hygine at all.

But sometimes we want to construct matching pattern using local variables:
```julia
using MacroTools
let 
  sym = :f_ # or `sym = :sin` will match
  @capture(:(sin(1)), $(sym)(a_))
end
```
then we need hygine, otherwise we can't access to the variable (in outer scope viewed from the `@capture` scope).

Reply:
Ok. I'm unsure about the fall-out from this, but since all tests pass, we'll see if this breaks anybody's code.